DeskAPIClient/DeskAPIClient/Models/DSAPIOpportunityFilter.m

```diff
@@ -0,0 +1,85 @@
+//     promote products derived from this software without specific prior written permission.
+//
+//  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED
+//  WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
+//  PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+//  ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED
+//  TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+//  HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+//  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+//  POSSIBILITY OF SUCH DAMAGE.
+//
+
+#import "DSAPIOpportunityFilter.h"
+
+#define kClassName @"opportunity_filter"
+
+@implementation DSAPIOpportunityFilter
+
++ (NSString *)className
+{
+    return kClassName;
+}
+
++ (NSURLSessionDataTask *)listFiltersWithParameters:(NSDictionary *)parameters
+                                             client:(DSAPIClient *)client
+                                              queue:(NSOperationQueue *)queue
+                                            success:(DSAPIPageSuccessBlock)success
+                                            failure:(DSAPIFailureBlock)failure
+{
+    return [super listResourcesAt:[DSAPIOpportunityFilter classLinkWithBaseURL:client.baseURL]
+                       parameters:parameters
+                           client:client
+                            queue:queue
+                          success:success
+                          failure:failure];
+}
+
+
+- (NSURLSessionDataTask *)showWithParameters:(NSDictionary *)parameters
+                                       queue:(NSOperationQueue *)queue
+                                     success:(void (^)(DSAPIOpportunityFilter *filter))success
+                                     failure:(DSAPIFailureBlock)failure
+{
+    return [super showWithParameters:parameters
+                               queue:queue
+                             success:^(DSAPIResource *resource) {
+                                 if (success) {
+                                     success((DSAPIOpportunityFilter *)resource);
+                                 }
+                             }
+                             failure:failure];
+}
+
+
+- (NSURLSessionDataTask *)listOpportunitiesWithParameters:(NSDictionary *)parameters
+                                                queue:(NSOperationQueue *)queue
+                                              success:(DSAPIPageSuccessBlock)success
+                                              failure:(DSAPIFailureBlock)failure
+{
+    return [self listResourcesForRelation:@"opportunities" // TODO: replace with [DSAPIOpportunity classNamePlural]
```

Comment:
"TODO" is addressed in https://github.com/forcedotcom/DeskApiClient-ObjC/pull/18